extensions/shared/plan-upgrade-notification.js

```diff
@@ -1,23 +1,74 @@
 /**
  * External dependencies
  */
+import apiFetch from '@wordpress/api-fetch';
 import { dispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 import '@wordpress/notices';
 import { parse as parseUrl } from 'url';
 
+/**
+ * Internal dependencies
+ */
+import getSiteFragment from './get-site-fragment';
+import { isSimpleSite } from './site-type-utils';
+
+/**
+ * Returns a URL where the current site's plan can be viewed from.
+ * [Relative to current domain for JP sites]
+ *
+ * @return {string} A URL where the current site plan is viewable.
+ */
+function getPlanUrl() {
+	if ( undefined !== typeof window && window.location ) {
+		if ( isSimpleSite() ) {
+			return `https://wordpress.com/plans/my-plan/${ getSiteFragment() }`;
+		}
+		// Potentially a JP site may have a wordpress root: https//foo.com/custom/wp/root
+		// Unlikely, but technically also possible: https//foo.com/custom/wp/wp-admin/root
+		return `${ window.location.protocol }//${ getSiteFragment().replace(
+			'::',
+			'/'
+		) }/wp-admin/admin.php?page=jetpack#/my-plan`;
+	}
+}
+
 /**
  * Shows a notification when a plan is marked as purchased
  * after redirection from WPCOM.
  */
+( async () => {
+	if ( undefined !== typeof window && window.location ) {
+		const { query } = parseUrl( window.location.href, true );
+
+		if ( query.plan_upgraded ) {
+			let planName = 'a paid plan';
 
-if ( undefined !== typeof window && window.location ) {
-	const { query } = parseUrl( window.location.href, true );
+			updatePlanNameFromApi: try {
+				// not updating if simple site
+				if ( isSimpleSite() ) {
+					break updatePlanNameFromApi;
+				}
 
-	if ( query.plan_upgraded ) {
-		dispatch( 'core/notices' ).createNotice(
-			'success',
-			__( 'Plan upgraded. You may use premium blocks now.', 'jetpack' )
-		);
+				const jetpackSiteInfo = await apiFetch( { path: '/jetpack/v4/site' } );
+				const data = JSON.parse( jetpackSiteInfo.data );
+
+				planName = `the ${ data.plan.product_name } plan.`;
```

Comment:
It's ok to have the plan name in variable but the surrounding sentence ("the" and "plan") has to be in `__()` so that it'll be possible for translators to translate, i.e. the sentence without the plan name and the one with plan name needs to occur separately like it was before.

Reply:
gotcha - thanks for clarifying all that! was a little cloudy

Reply:
yeah this was a little clumsyfully handled. should be good now